### DISCUSSION TITLE
Web browser on  a different host

### DISCUSSION THREAD

**Discussion by christophgil** [MannLabs/alphatims#152] [General]
Dear Sander,

My colleagues can currently work with the x2go connection, and there is no hurry.
For the long term, I would like  to make things more comfortable.
It would be ideal to drag-and-drop a timstof folder on a Windows  .bat
file which would open the local browser on the Windows desktop with the result,
while backend computation is performed  on a fast Linux.

This is what I got so far:
The following lines work from the linux desktop such that I can open the result in the browser of a Windows PC:
export BOKEH_ALLOW_WS_ORIGIN=our-linux-server:$my_port
/home/$USER/.conda/envs/alphatims_develop/bin/alphatims gui --bruker_raw_data /path-to-timsfolder.d --port  $my_port |

Setting BOKEH_ALLOW_WS_ORIGIN is crucial. I do not understand why I have to put the
hostname of the Linux and not the hostname of the Windows PC.

Running this line opens the web browser on the Linux Desktop, while it can be displayed on the Windows web browser at the same time.
Unsetting the DISPLAY variable such that the browser can not open, breaks it and results cannot be displayed

Theoretically, this could be used with a fake display with Xvfb where the firefox could run in.

It would be nicer to avoid the need for a web browser on the linux.  Or maybe using a lightweight console browser like  w3m instead of firefox?
Is the browser invoked via xdg-open?

Still many things to find out. ...

A completely different approach would be to use the X11 xserver Xming on the Windows and run the Linux firefox on the Windows display.  However, I was not able to open an
X-application under Xming, maybe it is because of our network.  The Windows PCs can see the Linux but not vice versa for security reasons.

Perhaps you have any ideas

Kind regards

Christoph

**Comment by swillems** [comment]
Dear Christoph,

It is great to hear that the gui development branch of AlphaTims already opened up a few new possibilities for you. As mentioned before, I have little to no expertise with web deployment and unfortunately cannot invest too much time on this particular issue which is probably more related to Windows/Linux communication than AlphaTims itself. If the issue turns out to be solvable from the AlphaTims source code rather than being a Linux/Windows communication issues, I am of course happy to update the gui development branch for you. 

Perhaps one useful suggestion that I can give is [MobaXTerm](https://mobaxterm.mobatek.net/) as a substitute for Xming/x2go. I have found it to be far more functional, flexible and intuitive in the past and it has spared me a bunch of headaches when working on remote Linux servers from Windows machines.

While I realize my reply might not be what you hoped for, I would be grateful if you could keep us up to date so that other users with similar setups can find inspiration through your experience.

All the best,
Sander
